src/app/clusters/groups-server/groups-server.cpp

```diff
@@ -83,6 +83,22 @@ static EmberAfStatus GroupAdd(FabricIndex fabricIndex, EndpointId endpointId, Gr
     GroupDataProvider * provider = GetGroupDataProvider();
     VerifyOrReturnError(nullptr != provider, EMBER_ZCL_STATUS_NOT_FOUND);
 
+    // Check if there are key set associated with the given GroupId
+    auto it       = provider->IterateGroupKeys(fabricIndex);
+    bool keyFound = false;
+    GroupDataProvider::GroupKey entry;
+    GroupDataProvider::KeySet keys;
```

Comment:
nit: move keys declaration lower, where it is actually used.

it seems like keys is just to check existence, so maybe some name like `existenceCheckKeys` would also work.

Reply:
I have move this logic to a separate function. I believe it looks cleaner now. And keys declaration is now next to it use.